### DISCUSSION TITLE
Feature Request: API filtering by license expiration date

### DISCUSSION THREAD

**Discussion by nCubed** [keygen-sh/community#35] [Ideas]
The current API allows for [license filtering](https://keygen.sh/docs/api/licenses/#licenses-list-filters) by `status` or `expiring` to retrieve licenses about to expire. However, Keygen considers a license to be in the expiring state only when the license is w/in three days of expiration.

We would like the ability to filter on licenses that are expiring within n-days. For example:

```
https://api.keygen.sh/v1/accounts/{ACCOUNT}/licenses?days-to-expiry=30
```

Any chance this type of filtering (or similar) is on the roadmap?

**Comment by ezekg** [comment]
We actually do have a new filter for this, but we forgot to add it to the docs (which I just added). 🙂

You can now use [the `expires` filter](https://keygen.sh/docs/api/licenses/#licenses-list-filters-expires) for this:

```
GET https://api.keygen.sh/v1/accounts/{ACCOUNT}/licenses?expires[in]=30d
```

**Reply by nCubed** [reply]
Excellent. Thanks getting those docs updated so quickly. Cheers!
